tests/test_protocols/test_abci/test_abci_fuzz.py

```diff
@@ -0,0 +1,359 @@
+
+    return type_node
+
+
+def create_abci_tree() -> Node:
+    """Create Tendermint ABCI types tree by unfolding fields"""
+
+    abci_types = descriptor_parser(tendermint.abci.types_pb2.DESCRIPTOR)
+    return {k: unfold_nested(v) for k, v in abci_types.items()}
+
+
+def create_tender_type_tree(type_tree: Node, abci_tree: Node) -> Node:
+    """Create type tree with Tendermint primitives (as string)"""
+
+    tender_type_tree = {}
+    for k in type_tree:
+        abci_node, type_node = abci_tree[k], type_tree[k]
+        replace_keys(abci_node, KEY_MAPPING.get(k, {}))  # type: ignore
+        tender_type_tree[k] = _translate(copy.deepcopy(type_node), abci_node)
+
+    return tender_type_tree
+
+
+# 3. Create hypothesis strategies
+def _init_subtree(node: Any) -> Any:
+    """Initialize subtree with hypothesis strategies"""
+
+    def init_repeated(repeated_type: Any) -> SearchStrategy:
+        """Repeated fields must be tuples for Tendermint protobuf"""
+        if isinstance(repeated_type, str) or repeated_type in PYTHON_PRIMITIVES:
+            strategy = STRATEGY_MAP[repeated_type]
+            return st.lists(strategy, min_size=0, max_size=100)
+        elif isinstance(repeated_type, tuple):
+            cls, kws = repeated_type
+            strategy = st.builds(cls, **_init_subtree(kws))
+            return st.lists(strategy, min_size=0, max_size=100)
+        else:
+            raise NotImplementedError(f"Repeated in {name}: {repeated_type}")
+
+    if isinstance(node, str):
+        return node
+
+    kwargs: Node = {}
+    for name, d_type in node.items():
+        if isinstance(d_type, str) or d_type in PYTHON_PRIMITIVES:
+            if name == "nanos":  # special case
+                kwargs[name] = STRATEGY_MAP[name]
+            else:
+                kwargs[name] = STRATEGY_MAP[d_type]
+        elif is_enum(d_type):
+            kwargs[name] = st.sampled_from(list(d_type))
+        elif isinstance(d_type, tuple):
+            container, content = d_type
+            if container is list:
+                kwargs[name] = init_repeated(content)
+            else:
+                kwargs[name] = st.builds(container, **_init_subtree(content))
+        else:
+            raise NotImplementedError(f"{name}: {d_type}")
+
+    return kwargs
+
+
+def init_type_tree_hypotheses(type_tree: Node) -> Node:
+    """
+    Initialize the hypothesis strategies
+
+    :param type_tree: mapping from message / field name to type.
+    :return: mapping from message / field name to initialized primitive.
+    """
+    return {k: _init_subtree(node) for k, node in type_tree.items()}
+
+
+def create_hypotheses() -> Any:
+    """Create hypotheses for ABCI Messages"""
+
+    aea_protocol, *_ = get_protocol_readme_spec()
+    speech_acts = aea_protocol["speech_acts"]
+
+    # 1. create type tree from speech acts
+    type_tree = create_aea_abci_type_tree(speech_acts)
+    type_tree.pop("dummy")  # TODO: known oddity on our side
+
```

Comment:
Why do we need this "dummy"?

Reply:
I don't think we need it. But I'm not just gonna change the protocol either.